docs/3_NEW_TEMPLATE.md

```diff
@@ -1,15 +1,406 @@
+    }
+
+    variable "randomization_level" {
+        description = "Number of additional random characters to include in resource names to insulate against unexpected resource name collisions."
+        type        = number
+        default     = 8
+    }
+
+    variable "resource_group_location" {
+        description = "The Azure region where all resources in this template should be created."
+        type        = string
+    }
+
+    // ---- Storage Account Configuration ----
+    variable "storage_account_tier" {
+        description = "Determines the level of performance required."
+        type = string
+        default = "Standard"
+    }
+
+    variable "storage_account_replication" {
+        description = "Defines the type of replication to use for this storage account. Valid options are LRS*, GRS, RAGRS and ZRS."
+        type = string
+        default = "LRS"
+    }
+
+    variable "storage_account_tags" {
+        description = "Metadata about the storage account created."
+        type = string
+        default = "hw-from-scratch"
+    }
+
+    // ---- Service Plan Module Configuration ----
+    variable "service_plan_tier" {
+        description = "The tier under which the service plan is created. Details can be found at https://docs.microsoft.com/en-us/azure/app-service/overview-hosting-plans."
+        type = string
+        default = "Dynamic"
+    }
+
+    variable "service_plan_size" {
+        description = "The compute and storage needed for the service plan to be deployed. Details can be found at https://azure.microsoft.com/en-us/pricing/details/app-service/windows/"
+        type = string
+        default = "Y1"
+    }
+
+    variable "service_plan_kind" {
+        description = "The kind of Service Plan to be created. i.e. FunctionApp"
+        type = string
+        default = "FunctionApp"
+    }
+    ```
+
+1. Open the commons.tf file and paste the the following:
+
+    ```HCL
+    module "provider" {
+        source = "../../modules/providers/azure/provider"
+    }
+
+    resource "random_string" "workspace_scope" {
+        keepers = {
+            # Generate a new id each time we switch to a new workspace or app id
+            ws_name = replace(trimspace(lower(terraform.workspace)), "_", "-")
+            app_id  = replace(trimspace(lower(var.name)), "_", "-")
+        }
+
+        length  = max(1, var.randomization_level) // error for zero-length
+        special = false
+        upper   = false
+    }
+
+    locals {
+        // sanitize names
+        app_id  = random_string.workspace_scope.keepers.app_id
+        region  = replace(trimspace(lower(var.resource_group_location)), "_", "-")
+        ws_name = random_string.workspace_scope.keepers.ws_name
+        suffix  = var.randomization_level > 0 ? "-${random_string.workspace_scope.result}" : ""
+
+        // base name for resources, name constraints documented here: https://docs.microsoft.com/en-us/azure/architecture/best-practices/naming-conventions
```

Comment:
There is a lot of context regarding randomizing resource names that seems strange *unless you have been bitten by it in the future*. It may be worthwhile to add context about why we do it

Reply:
Excellent point! Added brief comments about in Step 5 of step 5!